src/test_l1_dsDisplay.c

```diff
@@ -344,30 +328,28 @@ void test_l1_dsDisplay_positive_dsGetDisplay(void) {
 
     int result;
     intptr_t displayHandle1, displayHandle2;
-    size_t numPorts = sizeof(kSupportedPortTypes) / sizeof(kSupportedPortTypes[0]);
 
     // Step 01: Initialize the display sub-system
     result = dsDisplayInit();
     UT_LOG("\n In %s Return value: [%d]\n", __FUNCTION__, result);
     UT_ASSERT_EQUAL(result, dsERR_NONE);
 
-    for (size_t i = 0; i < numPorts; i++) {
-        dsVideoPortType_t vType = kSupportedPortTypes[i];
-        UT_LOG("\n In %s Port Type: [%d]\n", __FUNCTION__, vType);
-        // Step 02: Call dsGetDisplay() for each valid port
-        result = dsGetDisplay(vType, i, &displayHandle1);
-        UT_LOG("\n In %s Return value: [%d]\n", __FUNCTION__, result);
-        UT_ASSERT_EQUAL(result, dsERR_NONE);
-        UT_LOG("Display handle for port type %d: %ld\n", vType, (long)displayHandle1);
-
-        // Step 03: Call the last value again, and compare the results
-        result = dsGetDisplay(vType, i, &displayHandle2);
-        UT_LOG("\n In %s Return value: [%d]\n", __FUNCTION__, displayHandle1, displayHandle2);
-        UT_ASSERT_EQUAL(result, dsERR_NONE);
-        UT_LOG("\n In %s Comparison: [%d = %d]\n", __FUNCTION__, result);
-        UT_ASSERT_EQUAL(displayHandle1, displayHandle2);
-        UT_LOG("Repeated display handle for port type %d: %ld\n", vType, (long)displayHandle2);
-    }
+    uint32_t vType = UT_KVP_PROFILE_GET_UINT32("dsDisplay/Video_Ports");
```

Comment:
read Number_of_ports and loop through it. Read Video_Ports as array. Do the same in other functions

Reply:
done